src/test/java/com/demoqa/shop/steps/ProdPageStepDef.java

```diff
@@ -53,48 +55,47 @@ public void filterOnDefault() {
     public void allItemsDisplayed() {
         ProductsPage myPage = (ProductsPage) ScenarioContext.getInstance().getContext(Context.CURRENT_PAGE);
         ATFAssert.assertTrue("Not all items displayed", myPage.getAllElements().isDisplayed(), "All elements displayed");
-
+        log.info("All items are displayed");
     }
 
-    @When("the user selects a {string} color")
-    public void userSelectsColor(String color) {
-        color = "59";
-
+    @When("user selects the following filter:")
+    public void userSelectsColor(Map<String, String> filterMap) {
         ProductsPage myPage = (ProductsPage) ScenarioContext.getInstance().getContext(Context.CURRENT_PAGE);
-        myPage = new ProductsPage(Browser.getBrowser());
-        myPage.selectColor(color);
-    }
-
-    @Then("only the items with that color are displayed")
-    public void itemsWithSelectedColorDisplayed() {
-        ProductsPage page = (ProductsPage) ScenarioContext.getInstance().getContext(Context.CURRENT_PAGE);
-        page.getFisrstItem().click();
-        ATFAssert.assertTrue("No such color for that item", page.getItemColorBlack().isDisplayed(), "Color black available for this item");
+        if (filterMap.containsKey("color")) {
+            myPage.selectColor(filterMap.get("color"));
+            log.info("Color black is selected");
+        } else if (filterMap.containsKey("size")) {
+            myPage.selectSize(filterMap.get("size"));
+            log.info("Size 32 is selected");
+        }
     }
 
-    @When("the user selects a {string} size")
-    public void userSelectsSize(String size) {
-        size = "108";
+    @Then("only the items with that {string} are displayed")
+    public void itemsWithSelectedColorDisplayed(String filter) {
         ProductsPage page = (ProductsPage) ScenarioContext.getInstance().getContext(Context.CURRENT_PAGE);
-        page = new ProductsPage(Browser.getBrowser());
-        driver.get(page.getUrl());
-        page.selectSize(size);
-    }
 
-    @Then("only the items with that size are displayed")
-    public void itemsWithSelectedSizeDisplayed() {
-        ProductsPage page = (ProductsPage) ScenarioContext.getInstance().getContext(Context.CURRENT_PAGE);
-        page.getFisrstItem().click();
-        ATFAssert.assertTrue("No such size for that item", page.getItemSize32().isDisplayed(), "Size 32 available for this item");
+        switch (filter) {
+            case "color":
+                page.getFisrstItem().click();
+                wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//select[@id = 'pa_color']//option[@value='black']")));
```

Comment:
here you ca use public static ExpectedCondition<WebElement> visibilityOf(WebElement element)
it will be looks like  wait.until(ExpectedConditions.visibilityOf(page.getFisrstItem())

Reply:
yep, it makes sense, I'll make changes